arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/Keccak.java

```diff
@@ -78,18 +87,40 @@ private static int numKeccak(long x) {
   public void tracePreOpcode(final MessageFrame frame) {
     final OpCode opCode = this.hub.opCode();
 
-    // Capture calls to SHA3
-    if (opCode == OpCode.SHA3) {
-      if (frame.stackSize() > 1) {
-        final long sha3Size = Words.clampedToLong(frame.getStackItem(1));
-        this.sha3Sizes.peek().add(sha3Size);
+    final PlatformController pch = this.hub.pch();
+
+    if (pch.exceptions().none()) {
+      // Capture calls to SHA3.
+      if (opCode == OpCode.SHA3) {
+        callShakira(frame, 1, 0, this.sha3Sizes);
+      }
+
+      // Capture contract deployment
+      // TODO: compute the gas cost if we are under deployment.
+      if (opCode == OpCode.RETURN && hub.currentFrame().underDeployment()) {
+        callShakira(frame, 1, 0, this.deployedCodeSizes);
+      }
+
+      if (opCode == OpCode.CREATE2 && pch.aborts().none()) {
+        callShakira(frame, 2, 1, this.create2Sizes);
       }
     }
+  }
+
+  private void callShakira(
+      final MessageFrame frame,
+      final int codeSizeStackItemOffset,
+      final int codeOffsetStackItemOffset,
```

Comment:
On the stack the order is always  `offset, size` it would be more natural to order things accordingly in `callShakira`.